cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -76,7 +76,7 @@ func GetStaticPodSpecs(cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmap
 			ImagePullPolicy: v1.PullIfNotPresent,
 			Command:         getSchedulerCommand(cfg),
```

Comment:
IIRC, we don't explicitly set  `--port 0` to explicitly disable insecure serving.

If we do it would break insecure serving of metrics for the scheduler and controller-manager, that could be mitigated by an action required

Reply:
are you suggesting adding --port=0 for both with an action required as part of this PR?

the components are binding on localhost so they are not exposed to external consumers. but i can see that there may be cases where metrics are proxied to an external consumer.

the alternative is to keep the insecure serving supported and wait for the action-required to come from the components them self when --port is removed.

Reply:
That’s fine by me. I’ll open an issue to discuss secure defaults